Add Meson as additional (future?) build system

markuspg opened the pull request:
This pull request introduces Meson as additional build system. Since I'm from the "CMake camp" this is a best effort try only, but it compiles a working binary. The Gnome ecosystem seems to be strongly invested in Meson, so I guess this is the right way to go for GnoTime. Right now I do not intend to replace Autotools by Meson, it's more a prototype to allow early integration and testing.

markuspg committed:
Add Meson as additional (future?) build system

eli-schwartz commented on meson.build:
```diff
@@ -0,0 +1,52 @@
+  'GnoTime',
+  'c',
+  license: 'GPL-2.0-or-later',
+  meson_version: '>= 1.0.0',
```
This configures fine using Meson 0.49.0, dropping the minimum version to that would enable e.g. building on Debian buster (oldstable).

Seems like you already do support it, minus the version guard. :)

markuspg replied:
I lowered the required Meson version to `0.49.0`. I really don't know Meson well, so when I couldn't `meson compile` with `0.49.0` (generally we are aiming for compatibility with Debian oldstable) I went with `1.0.0` to be on the safe side.

eli-schwartz commented on src/meson.build:
```diff
@@ -0,0 +1,85 @@
+gnotime_cfg_data = configuration_data()
+gnotime_cfg_data.set_quoted('DATADIR', get_option('prefix') / get_option('datadir') / 'gnotime')
```
This macro is never used, as far as I can tell. There are three macros set to the same directory location, the other two are used though.

markuspg replied:
These macros are not being used by the code directly, but by some magic of the `libgnome` stuff. So for now I would prefer to keep them to unblock the integration of Meson, but I gladly drop them when we got rid of `libgnome`.

eli-schwartz commented on src/meson.build:
```diff
@@ -0,0 +1,85 @@
+gnotime_cfg_data.set_quoted('GNOMELOCALEDIR', get_option('prefix') / get_option('localedir'))
+gnotime_cfg_data.set_quoted('GTTDATADIR', get_option('prefix') / get_option('datadir') / 'gnotime')
+gnotime_cfg_data.set_quoted('GTTGLADEDIR', get_option('prefix') / get_option('datadir') / 'gnotime')
+gnotime_cfg_data.set_quoted('LIBDIR', get_option('prefix') / get_option('libdir'))
```
This macro is never used, as far as I can tell.

markuspg replied:
See above

eli-schwartz commented on src/meson.build:
```diff
@@ -0,0 +1,85 @@
+gnotime_cfg_data.set_quoted('GTTGLADEDIR', get_option('prefix') / get_option('datadir') / 'gnotime')
+gnotime_cfg_data.set_quoted('LIBDIR', get_option('prefix') / get_option('libdir'))
+gnotime_cfg_data.set_quoted('PACKAGE', 'gnotime')
+gnotime_cfg_data.set_quoted('PREFIX', get_option('prefix'))
+gnotime_cfg_data.set_quoted('SYSCONFDIR', get_option('sysconfdir'))
```
These two macros are never used, as far as I can tell.

markuspg replied:
See above

eli-schwartz commented on src/meson.build:
```diff
@@ -0,0 +1,85 @@
+  x11_dep,
+  xscrnsaver_dep,
+]
+if dbus_glib_dep.found()
+  dbus_glib_arg = 'WITH_DBUS=1'
+  gnotime_deps += dbus_glib_dep
+else
+  dbus_glib_arg = 'WITH_DBUS=0'
+endif
```
These need to be prefixed with -D.

markuspg replied:
Thank you for the hint => fixed

markuspg commented:
@eli-schwartz Unfortunately I don't know Meson well. If you volunteer I'd gladly take you on future pull requests extending the usage of Meson (e.g. for the schema files when switching to GSettings).

oskarb commented on src/meson.build:
```diff
@@ -0,0 +1,84 @@
+  'ghtml.c',
+  'ghtml-deprecated.c',
+  'idle-dialog.c',
+  'idle-timer.c',
```
Is this file actually included in the automake build? I noticed in my timeout work yesterday that changes in this file didn't seem to trigger a rebuild. If it's not included in the automake build, perhaps this matter also affects the cmake build?

Maybe we have more files like that?

markuspg replied:
Well spotted! I added both the `idle-timer`  header and source files to `src/Makefile.am`. For further missing files - I couldn't find any :-)

markuspg committed:
Fix idle-timer.(c|h) missing in src/Makefile.am

markuspg committed:
Ensure versions are aligned between all builders

Autotools was still on (even more) ancient versions than Ubuntu 14.04
"Trusty Tahr".
